v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java

```diff
@@ -173,123 +173,156 @@ public GcsIO(ProtegrityDataTokenizationOptions options) {
     this.options = options;
   }
 
-  public PCollection<? extends Serializable> read(Pipeline pipeline, SchemasUtils schema) {
-    if (options.getInputGcsFileFormat() == FORMAT.JSON) {
-      return pipeline
-          .apply("ReadJsonFromGCSFiles",
-              TextIO.read().from(options.getInputGcsFilePattern()));
-    } else if (options.getInputGcsFileFormat() == FORMAT.CSV) {
-      PCollectionTuple jsons = pipeline
-          /*
-           * Step 1: Read CSV file(s) from Cloud Storage using {@link CsvConverters.ReadCsv}.
-           */
-          .apply(
-              "ReadCsvFromGcsFiles",
-              CsvConverters.ReadCsv.newBuilder()
-                  .setCsvFormat(options.getCsvFormat())
-                  .setDelimiter(options.getCsvDelimiter())
-                  .setHasHeaders(options.getCsvContainsHeaders())
-                  .setInputFileSpec(options.getInputGcsFilePattern())
-                  .setHeaderTag(CSV_HEADERS)
-                  .setLineTag(CSV_LINES)
-                  .build())
-          /*
-           * Step 2: Convert lines to Json.
-           */
-          .apply(
-              "LineToJson",
-              CsvConverters.LineToFailsafeJson.newBuilder()
-                  .setDelimiter(options.getCsvDelimiter())
-                  .setJsonSchema(schema.getJsonBeamSchema())
-                  .setHeaderTag(CSV_HEADERS)
-                  .setLineTag(CSV_LINES)
-                  .setUdfOutputTag(PROCESSING_OUT)
-                  .setUdfDeadletterTag(PROCESSING_DEADLETTER_OUT)
-                  .build());
-
-      if (options.getNonTokenizedDeadLetterGcsPath() != null) {
+  private PCollection<String> readJson(Pipeline pipeline) {
+    return pipeline
+        .apply("ReadJsonFromGCSFiles",
+            TextIO.read().from(options.getInputGcsFilePattern()));
+  }
+
+  private PCollection<Row> readAvro(Pipeline pipeline, Schema beamSchema) {
+    org.apache.avro.Schema avroSchema = AvroUtils.toAvroSchema(beamSchema);
+    PCollection<GenericRecord> genericRecords = pipeline.apply(
+        "ReadAvroFiles",
+        AvroIO.readGenericRecords(avroSchema).from(options.getInputGcsFilePattern()));
+    return genericRecords
+        .apply(
+            "GenericRecordToRow", MapElements.into(TypeDescriptor.of(Row.class))
+                .via(AvroUtils.getGenericRecordToRowFunction(beamSchema)))
+        .setCoder(RowCoder.of(beamSchema));
```

Comment:
Let's merge these 2 parts into 1 the same way it is done in `writeAvro` function.